### DISCUSSION TITLE
Use sqlite-based cache by default, when available

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#35267] [Suggest an Idea]
### Todos

- [ ] Figure out way to deal with "install time compilation".
- [ ] Let Renovate use the sqlite-based cache, if available.

### Context

Rescuing an idea from getting lost in the Discussions. `@viceice` asked:

> should we switch to the sqlite based cache by default if sqlite is available? it's much faster.

And `@rarkins` responded:

> Yes but the install time compilation is unfortunately a problem for some

#### Related discussion

- #29767

**Comment by viceice** [comment]
sqlite is usually using a prebuild binary from GitHub releases

**Comment by rarkins** [comment]
@viceice @zharinov are we able to detect when the optional sqlite dependency has succeeded, and therefore us sqlite-first in such cases?

**Comment by viceice** [comment]
> @viceice @zharinov are we able to detect when the optional sqlite dependency has succeeded, and therefore us sqlite-first in such cases?

I think we should switch to the node built-in sqlite implementation if available. so we can drop the external dependency.


I can propose a PR for next renovate major which should be published soon anyways.

**Comment by rarkins** [comment]
Important to check the performance of the native implementation first

**Comment by viceice** [comment]
> Important to check the performance of the native implementation first 

should be good, it's native sqlite 

https://github.com/nodejs/node/tree/main/deps/sqlite

**Comment by HonkingGoose** [comment]
What's the status of this issue? Is there anything we need to do to consider this done?

We seem to use some kind of sqlite cache? See: [`lib/util/cache/package/sqlite.ts` on the `main` branch](https://github.com/renovatebot/renovate/blob/main/lib/util/cache/package/sqlite.ts).

Do we use the sqlite cache by default? If yes, we can probably close this issue? If no, we should explain what we still need to do. 😉

**Comment by viceice** [comment]
it's still masked behind an experimental environment variable

**Comment by rarkins** [comment]
I'm going to convert this to a Discussion. I don't think we have a compelling need for this just yet
